"kubectl cp" with no arguments causes a panic

<!-- This form is for bug reports and feature requests ONLY!

If you're looking for help check [Stack Overflow](https://stackoverflow.com/questions/tagged/kubernetes) and the [troubleshooting guide](https://kubernetes.io/docs/tasks/debug-application-cluster/troubleshooting/).

If the matter is security related, please disclose it privately via https://kubernetes.io/security/.
-->

**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>
/kind bug
> /kind feature


**What happened**:

```
# kubectl cp
panic: runtime error: index out of range

goroutine 1 [running]:
k8s.io/kubernetes/pkg/kubectl/cmd.(*CopyOptions).Run(0xc4201b1420, 0x2420990, 0x0, 0x0, 0x0, 0x0)
	/workspace/anago-v1.11.0-beta.2.141+8745ea56e3f1f3/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/cmd/cp.go:167 +0x483
k8s.io/kubernetes/pkg/kubectl/cmd.NewCmdCp.func1(0xc420a47b80, 0x2420990, 0x0, 0x0)
	/workspace/anago-v1.11.0-beta.2.141+8745ea56e3f1f3/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/cmd/cp.go:96 +0x93
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).execute(0xc420a47b80, 0x2420990, 0x0, 0x0, 0xc420a47b80, 0x2420990)
	/workspace/anago-v1.11.0-beta.2.141+8745ea56e3f1f3/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:760 +0x2c1
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc42036e280, 0xc4206423c0, 0x12a05f200, 0xc420a15ee8)
	/workspace/anago-v1.11.0-beta.2.141+8745ea56e3f1f3/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:846 +0x30a
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).Execute(0xc42036e280, 0x17ecbe0, 0x2402060)
	/workspace/anago-v1.11.0-beta.2.141+8745ea56e3f1f3/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:794 +0x2b
main.main()
	/workspace/anago-v1.11.0-beta.2.141+8745ea56e3f1f3/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kubectl/kubectl.go:50 +0x196
```

**What you expected to happen**:

I think `kubectl cp` should just show the help or something?

**How to reproduce it (as minimally and precisely as possible)**:

`kubectl cp`

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): `Client Version: version.Info{Major:"1", Minor:"11+", GitVersion:"v1.11.0-rc.1", GitCommit:"8745ea56e3f1f3ad20050c1762eb6ba6f7786675", GitTreeState:"clean", BuildDate:"2018-06-21T00:01:04Z", GoVersion:"go1.10.2", Compiler:"gc", Platform:"linux/amd64"}`

- Cloud provider or hardware configuration:
- OS (e.g. from /etc/os-release): ubuntu 16.04 amd64
- Kernel (e.g. `uname -a`): `Linux hex-32c1-pm 4.4.0-122-generic #146-Ubuntu SMP Mon Apr 23 15:34:04 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux`
- Install tools:
- Others:

`kubectl cp` with some arguments crash all the way from 1.9.0 to 1.10.5 (1.8.14 is ok):

Command: `kubectl cp file container:` (no trailing slash):

1.9.0:
```
panic: runtime error: index out of range

goroutine 1 [running]:
k8s.io/kubernetes/pkg/kubectl/cmd.copyToPod(0x2a6f640, 0xc4207038c0, 0xc4202ca480, 0x2a42480, 0xc42000c018, 0x2a42480, 0xc42000c020, 0x0, 0x0, 0x0, ...)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/cmd/cp.go:170 +0x888
k8s.io/kubernetes/pkg/kubectl/cmd.runCopy(0x2a6f640, 0xc4207038c0, 0xc4202ca480, 0x2a42480, 0xc42000c018, 0x2a42480, 0xc42000c020, 0xc420112800, 0x2, 0x4, ...)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/cmd/cp.go:137 +0x23b
k8s.io/kubernetes/pkg/kubectl/cmd.NewCmdCp.func1(0xc4202ca480, 0xc420112800, 0x2, 0x4)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/cmd/cp.go:72 +0x89
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).execute(0xc4202ca480, 0xc4201127c0, 0x4, 0x4, 0xc4202ca480, 0xc4201127c0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:603 +0x234
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc42031d200, 0x8000104, 0x0, 0xffffffffffffffff)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:689 +0x2fe
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).Execute(0xc42031d200, 0xc4207038c0, 0x2a42440)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:648 +0x2b
k8s.io/kubernetes/cmd/kubectl/app.Run(0x0, 0x0)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kubectl/app/kubectl.go:41 +0xd5
main.main()
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kubectl/kubectl.go:27 +0x26
```

1.10.5:
```
panic: runtime error: index out of range

goroutine 1 [running]:
k8s.io/kubernetes/pkg/kubectl/cmd.copyToPod(0x23b2840, 0xc42067bc80, 0xc420824c80, 0x2386d40, 0xc42008a008, 0x2386d40, 0xc42008a010, 0x0, 0x0, 0x0, ...)
	/workspace/anago-v1.10.5-beta.0.53+32ac1c9073b132/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/cmd/cp.go:171 +0x888
k8s.io/kubernetes/pkg/kubectl/cmd.runCopy(0x23b2840, 0xc42067bc80, 0xc420824c80, 0x2386d40, 0xc42008a008, 0x2386d40, 0xc42008a010, 0xc420374080, 0x2, 0x4, ...)
	/workspace/anago-v1.10.5-beta.0.53+32ac1c9073b132/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/cmd/cp.go:138 +0x23b
k8s.io/kubernetes/pkg/kubectl/cmd.NewCmdCp.func1(0xc420824c80, 0xc420374080, 0x2, 0x4)
	/workspace/anago-v1.10.5-beta.0.53+32ac1c9073b132/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubectl/cmd/cp.go:73 +0x89
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).execute(0xc420824c80, 0xc420331040, 0x4, 0x4, 0xc420824c80, 0xc420331040)
	/workspace/anago-v1.10.5-beta.0.53+32ac1c9073b132/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:757 +0x2c1
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc42013d900, 0xc4208d7ed8, 0xc4208d7ee0, 0xc4204b8180)
	/workspace/anago-v1.10.5-beta.0.53+32ac1c9073b132/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:843 +0x334
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).Execute(0xc42013d900, 0x1900880, 0x2552520)
	/workspace/anago-v1.10.5-beta.0.53+32ac1c9073b132/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:791 +0x2b
main.main()
	/workspace/anago-v1.10.5-beta.0.53+32ac1c9073b132/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/cmd/kubectl/kubectl.go:50 +0x196
```

--- a/pkg/kubectl/cmd/cp.go
+++ b/pkg/kubectl/cmd/cp.go
@@ -164,6 +164,9 @@ func (o *CopyOptions) Validate(cmd *cobra.Command, args []string) error {
 }
 
 func (o *CopyOptions) Run(args []string) error {
+	if len(args) < 2 {
+		return fmt.Errorf("source and destination are required")
+	}
 	srcSpec, err := extractFileSpec(args[0])
 	if err != nil {
 		return err

